src/components/NavBar.tsx

```diff
@@ -2,13 +2,18 @@ import { Link } from "react-router-dom";
 import { useState } from "react";
 import { IoBagHandleOutline } from "react-icons/io5";
 import { IoPersonOutline } from "react-icons/io5";
-import "../styles/navbar.css";
 import { useBasket, BasketProduct } from "../context/useBasket";
+import { useAuth } from "../context/useAuth";
+import "../styles/navbar.css";
+import Login from "./Login";
+import Logout from "./Logout";
 
 const NavBar = () => {
-  const [open, setOpen] = useState(false);
+  const [open, setOpen] = useState<boolean>(false);
+  const [userMenu, setUserMenu] = useState<boolean>(false);
```

Comment:
I would change this to a more boolean like name for example `isUserMenuOpen` or `isUserMenuShown`.